src/sdf_xarray/__init__.py

```diff
@@ -253,9 +253,19 @@ def acquire_context(self, needs_lock=True):
     def load(self):  # noqa: PLR0912, PLR0915
         # Drop any requested variables
         if self.drop_variables:
+            # Build a mapping from underscored names to real variable names
+            name_map = {_rename_with_underscore(var): var for var in self.ds.variables}
+
             for variable in self.drop_variables:
-                # TODO: nicer error handling
-                self.ds.variables.pop(variable)
+                key = _rename_with_underscore(variable)
+                original_name = name_map.get(key)
+
+                if original_name:
+                    self.ds.variables.pop(original_name)
+                else:
+                    raise KeyError(
+                        f"Variable '{variable}' not found (interpreted as '{key}')."
+                    )
```

Comment:
I usually recommend writing these guard conditionals the other way around:

```suggestion
                if original_name is None:
                    raise KeyError(
                        f"Variable '{variable}' not found (interpreted as '{key}')."
                    )
                self.ds.variables.pop(original_name)              
```

Then the happy path is less indented and it's clearer what the error condition is

Reply:
That's definitely a lot cleaner, thanks for the idea